CMakeLists.txt

```diff
@@ -32,6 +32,28 @@ if (CYGWIN)
   set(CMAKE_FIND_LIBRARY_SUFFIXES ".dll")
 endif()
 
+find_package(Git REQUIRED)
+add_custom_command(
+  OUTPUT "${CMAKE_BINARY_DIR}/version_gen.h.tmp"
+  COMMAND "${CMAKE_COMMAND}" -E echo_append "#define ALIVE_VERSION_MACRO " > "${CMAKE_BINARY_DIR}/version_gen.h.tmp"
+  COMMAND "${GIT_EXECUTABLE}" describe --tags --dirty >> "${CMAKE_BINARY_DIR}/version_gen.h.tmp"
```

Comment:
`git describe` adds newline at the end, and storing result into a macro seems like the least ugly solution to end up with a version string with no newline at the end